packages/zwave-js/src/lib/commandclass/BarrierOperatorCC.ts

```diff
@@ -0,0 +1,267 @@
+import type { MessageOrCCLogEntry } from "@zwave-js/core";
+import {
+	CommandClasses,
+	enumValuesToMetadataStates,
+	parseBitMask,
+	validatePayload,
+	ValueMetadata,
+	ZWaveError,
+	ZWaveErrorCodes,
+} from "@zwave-js/core";
+import type { Driver } from "../driver/Driver";
+import {
+	CCCommand,
+	CCCommandOptions,
+	ccValue,
+	ccValueMetadata,
+	CommandClass,
+	commandClass,
+	CommandClassDeserializationOptions,
+	expectedCCResponse,
+	gotDeserializationOptions,
+	implementedVersion,
+} from "./CommandClass";
+// All the supported commands
+export enum BarrierOperatorCommand {
+	Set = 0x01,
+	Get = 0x02,
+	Report = 0x03,
+	CapabilitiesGet = 0x04,
+	CapabilitiesReport = 0x05,
+	EventSet = 0x06,
+	EventGet = 0x07,
+	EventReport = 0x08,
+}
+
+/**
+ * @publicAPI
+ */
+export enum BarrierState {
+	Closed = 0x00,
+	Closing = 0xfc,
+	Stopped = 0xfd,
+	Opening = 0xfe,
+	Open = 0xff,
+}
+
+// @publicAPI
+export enum SignalType {
+	Audible = 0x01,
+	Visual = 0x02,
+}
+
+// @publicAPI
+export enum SignalState {
+	OFF = 0x00,
+	ON = 0xff,
+}
+
+@commandClass(CommandClasses["Barrier Operator"])
+@implementedVersion(1)
+export class BarrierOperatorCC extends CommandClass {
+	declare ccCommand: BarrierOperatorCommand;
+}
+
+interface BarrierOperatorCCSetOptions extends CCCommandOptions {
+	targetValue: boolean;
+}
+
+@CCCommand(BarrierOperatorCommand.Set)
+export class BarrierOperatorCCSet extends BarrierOperatorCC {
+	public constructor(
+		driver: Driver,
+		options:
+			| CommandClassDeserializationOptions
+			| BarrierOperatorCCSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			this.targetValue = options.targetValue;
+		}
+	}
+
+	public targetValue: boolean;
```

Comment:
Thinking about it again, this property should have the same name and type as the corresponding one in the report (`barrierState`). You can specify the type `BarrierState.Open | BarrierState.Closed` on `BarrierOperatorCCSetOptions` to ensure only those get used.

Reply:
I would however rename `barrierState` to `state` and `barrierPosition` to `position`.

Reply:
changed all:
BarrierState -> State
(_)barrierPosition -> (_)position
(_)barrierState -> (_)state

shorter names